text/0000-opencensus-migration-plan.md

```diff
@@ -0,0 +1,141 @@
+# OpenCensus To OpenTelemetry Migration Plan
+
+Define the process of migrating applications, libraries, and agents from [OpenCensus](https://opencensus.io/) to
+OpenTelemetry.
+
+## Motivation
+
+While OpenTelemetry has a specification for an
+[OpenCensus trace bridge](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.12.0/specification/compatibility/opencensus.md),
+it lacks a broader context on how such bridges should be used to perform a migration from
+OpenCensus to OpenTelemetry. Additionally, OpenTelemetry should provide guidance to libraries
+currently using OpenCensus on how to migrate instrumentation to OpenTelemetry.
+
+## Explanation
+
+Migrating from OpenCensus to OpenTelemetry may require breaking changes to the telemetry produced
+because of:
+
+* Differences in semantic conventions for names and attributes
+* Data model differences
+* Instrumentation API feature differences
+* Differences between equivalent OC and OTel exporters
+
+Requirements:
+
+* Give application developers control over _when_ breaking changes are introduced.
+* No "long tail" of breaking changes to users when migrating.
+* Make it possible to migrate instrumentation without breaking the resulting telemetry.
+* Library maintainers should not need to maintain OpenCensus and OpenTelemetry instrumentation in parallel.
+
+This proposed process groups most breaking changes into the installation of the bridge. This gives
+users control over introducing the initial breaking change, and makes subsequent migrations of
+instrumentation (including in third-party libraries) non-breaking.
+
+## Migration Plans
+
+### Migrating from the OC Agent and Protocol
+
+Starting with a deployment of the OC agent, using the OC protocol, migrate by:
+
+1. Deploy the OpenTelemetry collector with OpenCensus and OTLP receivers and equivalent processors and exporters.
+2. **breaking**: For each workload sending the OC protocol, change to sending to the OpenTelemetry collector's OpenCensus receiver.
+3. Remove the deployment of the OC Agent
+4. For each workload, migrate the application from OpenCensus to OpenTelemetry, following the guidance below, and use the OTLP exporter
+
+### Requirements for Metrics and Trace bridges
+
+Bridges:
+
+* MUST NOT require OpenCensus to depend on OpenTelemetry
+* MUST require few or no changes to OpenCensus
+* MUST convert OpenCensus semantic conventions to OpenTelemetry semantic conventions for metric and span names and attributes
```

Comment:
Which semantic conventions are in scope? Are OTel semantic conventions stable enough to target? (e.g., would a bridge need to keep adding or changing based on new versions of semantic conventions?)

Reply:
All documented OpenCensus semantic conventions are in-scope, which includes HTTP and gRPC.  I've specifically included those.  I agree a bridge cannot be marked stable until the conventions it depends on are marked stable.